src/com/company/Main.java

```diff
@@ -1,19 +1,73 @@
 package com.company;
 
-import java.util.Scanner;
+import java.security.spec.RSAOtherPrimeInfo;
+import java.util.ArrayList;
+import java.util.List;
 
 public class Main {
 
+    private static final char MATCHING_CHAR = 'o';
+
     public static void main(String[] args) throws Exception {
-	// write your code here
-       /* Matrix spaceInvaderA = new Matrix(11,8);
-        char[][] invaderA = spaceInvaderA.getM_matrix();*/
-        String invaderALocation = "/Users/nikoladamjanovic/Downloads/invaderA";
+        final String invaderALocation = "/Users/nikoladamjanovic/Downloads/invaderA";
         MatrixParser parser = new MatrixParser(invaderALocation);
-        parser.printMatrix();
-        //String invaderA = parser.readFileAsString();
-        //System.out.println(invaderA);
+        char[][] invaderA = parser.getFilledMatrix();
+        //printMatrix(invaderA);
+
+        final String inputLocation = "/Users/nikoladamjanovic/Downloads/inputMatrix";
+        MatrixParser parserA = new MatrixParser(inputLocation);
+        char[][] searchMatrix = parserA.getFilledMatrix();
+        //printMatrix(searchMatrix);
+
+        processMatrix(invaderA, searchMatrix);
+    }
+
+    private static void processMatrix(char[][] invaderA, char[][] searchMatrix) {
+        List<Character> exactMatches = new ArrayList<Character>();
+        List<Character> invaderOnly = new ArrayList<Character>();
+        List<Character> searchOnly = new ArrayList<Character>();
+        int totalInvaders = 0;
+
+        for (int i = 0; i < searchMatrix.length - invaderA.length + 1; i ++) {
+            for (int j = 0; j < searchMatrix[0].length - invaderA[0].length + 1; j++) {
+                for (int k = 0; k < invaderA.length; k++) {
+                    for (int p = 0; p < invaderA[0].length; p++) {
 
+                        char invaderMatrixChar = invaderA[k][p];
+                        char searchMatrixChar = searchMatrix[i + k][j + p];
+
+                        if ((searchMatrixChar == MATCHING_CHAR) && (invaderMatrixChar == MATCHING_CHAR)){
```

Comment:
Could this be just _(searchMatrixChar == MATCHING_CHAR && invaderMatrixChar == MATCHING_CHAR)_